src/types.ts

```diff
@@ -29,8 +29,9 @@ export declare class ILogicRule {
   resolve(parent, args, ctx, info, options: IOptions): Promise<IRuleResult>
 }
 
+export type ICacheFunction = (parent: any, args: any) => string
```

Comment:
2nd parameter should have shape matching object-hash `hash()` [option](https://www.npmjs.com/package/object-hash#hashvalue-options)